### PR TITLE
lsp: add the diagnostic sign priority

### PR DESCRIPTION
**Pull request by glepnir**
@tjdevries  When we have multiple signs in col like `vim-gigutter vim-signfy`, we want the  priority of diagnosis sign is higher than other signs

### CODE REVIEW

File: runtime/lua/vim/lsp/util.lua [hunk 1]
```diff
@@ -1258,7 +1258,7 @@ do
   --- </pre>
   function M.buf_diagnostics_signs(bufnr, diagnostics)
     for _, diagnostic in ipairs(diagnostics) do
-      vim.fn.sign_place(0, sign_ns, diagnostic_severity_map[diagnostic.severity], bufnr, {lnum=(diagnostic.range.start.line+1)})
+      vim.fn.sign_place(0, sign_ns, diagnostic_severity_map[diagnostic.severity], bufnr, {lnum=(diagnostic.range.start.line+1),priority=90})
```
Comment (bew) [inline runtime/lua/vim/lsp/util.lua:1261]: `90` is pretty arbitrary, I think this should be a configurable value, idn't it?

Comment (glepnir) [reply runtime/lua/vim/lsp/util.lua:1261]: Yes, so this is a draft. The maintainer is required to determine the name of the variable

**Comment by tjdevries** [comment]
Hmmm, this is difficult to figure out where we should put the configuration.

Do you have any thoughts about where it makes the most sense? I will think about it some more.

**Comment by glepnir** [comment]
@tjdevries  I feel very confused about the current `lsp` directory structure. There is no good place to define this variable. But I think it is necessary to add a variable.  Maybe we can temporarily add it to the top of `M.buf_diagnostics_signs`. When your diagnosis pr is merged, we will find a suitable place and then modify it?

**Comment by tjdevries** [comment]
Yeah, we don't have a place currently for configuration really. I think for now, your suggestion of just assigning a local variable somewhere and using it in place seems quite good.

As you mention, perhaps my diagnostic PR, we can then expose it in an elegant way (probably as an option to the caller or something).

**Comment by glepnir** [comment]
Yes, I think your diagnostic pr should include this option. As a temporary configuration. we can do this. `9999999` set it to max.
```lua
local diagnostic_sign_priority = vim.g.lsp_diagnositc_sin_priority or 9999999
```

**Comment by tjdevries** [comment]
I implemented this in https://github.com/neovim/neovim/commit/f75be5e9d510d5369c572cf98e78d9480df3b0bb Thanks @glepnir but I ended up just doing it there to show how we can make new configuration much more easily.

**Pull request closed without merging**